lightning/src/util/sweep.rs

```diff
@@ -510,10 +608,17 @@ where
 			},
 		};
 
-		// As we didn't modify the state so far, the same filter_fn yields the same elements as
-		// above.
-		let respend_outputs = sweeper_state.outputs.iter_mut().filter(|o| filter_fn(&**o));
-		for output_info in respend_outputs {
+		// Watch outputs and update status.
+		for output_info in respend_descriptors {
+			let output_info =
+				match sweeper_state.outputs.iter_mut().find(|o| o.descriptor == *output_info) {
+					Some(output_info) => output_info,
+					None => {
+						// Output was already removed from the state.
```

Comment:
This is a new possibility, but can revert this once we re-retrieve the descriptors I think.